src/net_processing.cpp

```diff
@@ -403,6 +405,10 @@ limitedmap<uint256, int64_t> g_already_asked_for GUARDED_BY(cs_main)(MAX_INV_SZ)
 /** Map maintaining per-node state. */
 static std::map<NodeId, CNodeState> mapNodeState GUARDED_BY(cs_main);
 
+/** In-memory cache of all BIP157 compact filter checkpoints for the active chain. */
+static std::vector<std::pair<const CBlockIndex*, uint256>> active_chain_cf_headers;
+static boost::shared_mutex active_chain_cf_headers_mtx;
```

Comment:
https://github.com/bitcoin/bitcoin/pull/16442/commits/3499e8877b3be1a629647353ddb8bb380c10a513

Might want to make a note here about why we're adding more boost usage instead of using the utilities in `sync` (ie `std::mutex`). I assume this is because we want to allow multiple threads to be able to read the headers cache simultaneously?

Reply:
Yeah, the rationale is that `std` doesn't get `shared_mutex` until C++17. I'll add a comment.